src/lib/InstantSearch.ts

```diff
@@ -587,6 +588,20 @@ Feel free to give us feedback on GitHub: https://github.com/algolia/instantsearc
 
     this.mainHelper.clearCache().search();
   }
+
+  public sendEventToInsights(event: InsightsEvent) {
+    const insights = this.getInsightsMiddleware();
+    if (!insights) {
+      return;
+    }
+    insights.sendEvent(event);
+  }
+
+  private getInsightsMiddleware() {
+    return this.middleware.find(
```

Comment:
This doesn't seem like the most efficient method, I think it should be added to an object instead of an array